homeassistant/components/climate/systemair_savecair.py

```diff
@@ -0,0 +1,282 @@
+    SUPPORT_FAN_MODE, SUPPORT_ON_OFF, SUPPORT_OPERATION_MODE,
+    SUPPORT_TARGET_TEMPERATURE, ClimateDevice)
+from homeassistant.const import (
+    ATTR_TEMPERATURE, CONF_NAME, CONF_PASSWORD, TEMP_CELSIUS, TEMP_FAHRENHEIT)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['python-systemair-savecair==1.0.1']
+
+ATTR_SUPPLY_AIR_TEMPERATURE = "supply_air_temperature"
+ATTR_SUPPLY_AIR_FAN_SPEED = "supply_air_fan_speed"
+ATTR_EXTRACT_AIR_FAN_SPEED = "extract_air_fan_speed"
+
+FAN_OFF = "Off"
+FAN_LOW = "Low"
+FAN_NORMAL = "Medium"
+FAN_HIGH = "High"
+
+FAN_LIST = [FAN_OFF, FAN_LOW, FAN_NORMAL, FAN_HIGH]
+
+OPERATION_LIST = [STATE_AUTO, STATE_MANUAL]
+
+SUPPORT_FLAGS = SUPPORT_TARGET_TEMPERATURE | \
+                SUPPORT_FAN_MODE | \
+                SUPPORT_OPERATION_MODE | \
+                SUPPORT_AWAY_MODE | \
+                SUPPORT_ON_OFF
+
+SAVECAIR_FAN_MODES = {
+    1: FAN_OFF,
+    2: FAN_LOW,
+    3: FAN_NORMAL,
+    4: FAN_HIGH
+}
+
+SAVECAIR_OPERATION_MODES = {
+    STATE_AUTO: STATE_AUTO,
+    STATE_IDLE: STATE_IDLE,
+    STATE_MANUAL: STATE_MANUAL,
+}
+
+CONF_IAM_ID = "iam_id"
+DEFAULT_NAME = "SystemAIR"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Required(CONF_IAM_ID): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string
+})
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the SystemAIRplatform."""
+    name = config[CONF_NAME]
+    iam_id = config[CONF_IAM_ID]
+    password = config[CONF_PASSWORD]
+
+    # Create savecair client
+    from systemair.savecair import SavecairClient
+    client = SavecairClient(iam_id, password, loop=hass.loop, retry=True)
+    climate = SystemAIRClimate(client, name)
+    async_add_entities([climate])
+
+
+class SystemAIRClimate(ClimateDevice):
+    """Representation of the climate sensor."""
+
+    def __init__(self, client, name):
+        """Construct the SystemAIR Climate Device."""
+        self._name = name
+        self._client = client
+        self._client.start()
+
+    async def async_update(self):
+        """Retrieve latest state."""
+        await self._client.update_sensors()
+
+    async def async_added_to_hass(self):
+        """Register update signal handler."""
+        async def async_update_state():
+            self.async_schedule_update_ha_state()
+        self._client.on_update.append(async_update_state)
+        self._client.on_connect.append(async_update_state)
+
+    @property
+    def device_state_attributes(self):
+        """Update state attributes with additional attributes."""
+        attributes = {
+            ATTR_SUPPLY_AIR_TEMPERATURE: self.supply_air_temperature,
+            ATTR_SUPPLY_AIR_FAN_SPEED: self.extract_air_fan_speed,
+            ATTR_EXTRACT_AIR_FAN_SPEED: self.supply_air_fan_speed
+        }
+        attributes.update(self._client.data)
```

Comment:
`_client.data` still contains the relative timestamp `components_filter_time_left` according to the source on github:
https://github.com/perara/python-systemair-savecair/blob/master/systemair/savecair/parameters.txt#L4

Please don't update the state attributes with all the client data. Add specific keys here so that it's clear what attributes are getting exposed.

Reply:
Would you prefer this as a static timestamp, or a static date object?

Reply:
```
{
  "current_temperature": 25.6,
  "min_temp": 7,
  "max_temp": 35,
  "temperature": 28,
  "target_temp_step": 1,
  "current_humidity": 21,
  "fan_mode": "Medium",
  "fan_list": [
    "Off",
    "Low",
    "Medium",
    "High"
  ],
  "operation_mode": "manual",
  "operation_list": [
    "auto",
    "manual"
  ],
  "away_mode": "off",
  "supply_air_temperature": 19.6,
  "supply_air_fan_speed": 2520,
  "extract_air_fan_speed": 2520,
  "alarm_emt_state": false,
  "rh_sensor": 21,
  "user_mode_fireplace_supply": "5",
  "alarm_extra_controller_state": false,
  "control_regulation_airflow_manual_fan_stop": true,
  "alarm_oat_state": false,
  "function_active_heat_recovery": true,
  "function_active_moisture_transfer": false,
  "alarm_external_stop_state": false,
  "function_active_service_user_lock": false,
  "control_regulation_demand_co2_status": false,
  "user_mode_refresh_extract": "5",
  "function_active_defrosting": false,
  "function_active_cooker_hood": false,
  "alarm_pdm_rhs_state": false,
  "alarm_eat_state": false,
  "alarm_ect_state": false,
  "alarm_eft_state": false,
  "alarm_manual_mode_state": false,
  "main_user_mode": "1",
  "fan_log_request_reset": 1543310659,
  "alarm_overheat_temperature_state": false,
  "alarm_co2_state": false,
  "user_lock": false,
  "eco_mode": true,
  "alarm_filter_state": false,
  "week_schedule_any_defined": true,
  "alarm_oht_state": false,
  "week_schedule_lock": false,
  "alarm_sat_state": false,
  "control_regulation_demand_rh_status": false,
  "alarm_eaf_rpm_state": false,
  "function_active_cooling": false,
  "function_active_cooling_recovery": false,
  "supply_air_temp": 196,
  "main_airflow": "3",
  "control_regulation_airflow_type": "rpm",
  "outdoor_air_temp": 0,
  "alarm_frost_prot_state": false,
  "alarm_saf_ctrl_state": false,
  "user_mode_holiday_extract": "1",
  "alarm_rgs_state": "returned",
  "function_active_secondary_air": false,
  "user_mode_crowded_extract": "5",
  "user_mode_holiday_supply": "1",
  "digital_input_tacho_saf_value": 2520,
  "alarm_rat_state": false,
  "alarm_low_sat_state": false,
  "main_temperature_offset": 280,
  "user_mode_crowded_supply": "5",
  "user_mode_fireplace_extract": "1",
  "user_mode_auto_supply": "4",
  "function_active_heating": false,
  "alarm_fpt_state": false,
  "main_iaq": "economic",
  "user_mode_refresh_supply": "5",
  "alarm_rh_state": false,
  "alarm_secondary_air_state": false,
  "filter_lock": false,
  "digital_input_tacho_eaf_value": 2520,
  "components_filter_time_left": 1569238686,
  "alarm_bys_state": false,
  "alarm_eaf_ctrl_state": false,
  "alarm_defrosting_state": false,
  "alarm_manual_fan_stop_state": false,
  "pdm_input_temp_value": 256,
  "function_active_heater_cooldown": false,
  "function_active_free_cooling": false,
  "alarm_saf_rpm_state": false,
  "user_mode_away_supply": "1",
  "function_active_vacuum_cleaner": false,
  "control_regulation_airflow_unit_man_perc": null,
  "control_regulation_temp_unit": "celsius",
  "alarm_pdm_eat_state": false,
  "user_mode_away_extract": "1",
  "friendly_name": "System AIR",
  "supported_features": 5313
}
```

Reply:
2520 is two totally different values. One is suppy fan and one is extract fan.

Reply:
Yes, and both are duplicated, so there are four values instead of two.